src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -574,7 +574,7 @@ private void MakeConstantTuple(LocalSymbol inProgress, BoundExpression boundInit
 
             internal override ConstantValue GetConstantValue(SyntaxNode node, LocalSymbol inProgress, BindingDiagnosticBag diagnostics = null)
             {
-                if (this.IsConst && inProgress == this)
+                if (this.IsConst && (inProgress == this || this.ForbiddenZone?.Contains(node) == true))
```

Comment:
This will catch a direct reference to `x` in the initializer for `const x`. Is there also a problem with an indirect reference: `const int x = y; const int y = ...x...;`?

Reply:
Could not reproduce that failing even today. Added tests.